pkg/currency/currency.go

```diff
@@ -0,0 +1,42 @@
+package currency
+
+import (
+	"errors"
+	"fmt"
+	"math"
+)
+
+// Picodollar is a type to represent currency with 12 decimal precision
+type PicoDollar int64
+
+const (
+	PicoDollarsPerDollar = 1e12
+)
+
+// FromDollars converts a dollar amount (as a float) to Picodollars
+// This should mostly be used for testing, and real usage should be done purely in PicoDollars
+func FromDollars(dollars float64) (PicoDollar, error) {
+	if math.IsNaN(dollars) || math.IsInf(dollars, 0) {
+		return 0, errors.New("invalid dollar amount: must be a finite number")
+	}
+
+	picodollars := dollars * PicoDollarsPerDollar
+	if picodollars > math.MaxInt64 || picodollars < math.MinInt64 {
+		return 0, errors.New("dollar amount too large to convert to Picodollars")
+	}
+	return PicoDollar(picodollars), nil
+}
+
+// ToDollars converts PicoDollars to a dollar amount (as a float)
+func (p PicoDollar) toDollars() float64 {
+	return float64(p) / PicoDollarsPerDollar
+}
+
+// ToMicroDollars converts PicoDollars to MicroDollars (1e6 units per dollar)
+func (p PicoDollar) ToMicroDollars() int64 {
+	return int64(p / 1e6)
+}
+
+func (p PicoDollar) String() string {
+	return fmt.Sprintf("%.12f", p.toDollars())
+}
```

Comment:
that float conversion is not exact and reading a string that has 10+ spots is virtually impossible. I would just display picodollars as is. I would rather read `10032` than `0.000000000010032`

Reply:
IDK. They're both hard to read, but I think it's helpful to have some number that has a sense of scale to it. Otherwise we're going to be reaching for the calculator app every time we see a log line with picodollars in it.

Reply:
For context, we are going to have bigger numbers that get expressed in PicoDollars. Like the total spend for a Payer in a report. I'd much rather see that logged as `20.472123` than `2047212300000000` and have to convert it myself